src/test/scala/is/hail/methods/LinearRegressionBurdenSuite.scala

```diff
@@ -0,0 +1,248 @@
+
+                Position
+          1 2 3 4 5 6 7 8 9 10
+    Gene1 -
+    Gene2   -
+    Gene3     -
+    */
+
+    val vdsBurdenNoOverlap: VariantDataset = hc.importVCF("src/test/resources/regressionLinear.vcf")
+      .annotateVariantsIntervals("src/test/resources/regressionLinearNoOverlap.interval_list", "va.gene")
+      .annotateVariantsExpr("va.weight = v.start.toDouble")
+      .annotateVariantsExpr("""va.genes2 = if (isDefined(va.gene)) [va.gene] else range(0).map(x => "")""")
+      .annotateVariantsExpr("va.genes3 = if (isDefined(va.gene)) [va.gene] else NA: Array[String]")
+      .annotateVariantsExpr("va.genes4 = if (isDefined(va.gene)) [va.gene, va.gene].toSet else NA: Set[String]")
+      .annotateVariantsExpr("va.genes5 = if (isDefined(va.gene)) [va.gene, va.gene] else NA: Array[String]")
+      .annotateVariantsExpr("va.genes6 = NA: Set[String]")
+      .annotateVariantsExpr("va.genes7 = v.start")
+      .annotateVariantsExpr("va.genes8 = v.start.toDouble")
+      .annotateSamplesTable("src/test/resources/regressionLinear.pheno",
+        "Sample",
+        root = Some("sa.pheno"),
+        config = TextTableConfiguration(types = Map("Pheno" -> TDouble), missing = "0"))
+      .annotateSamplesTable("src/test/resources/regressionLinear.cov",
+        "Sample",
+        root = Some("sa.cov"),
+        config = TextTableConfiguration(types = Map("Cov1" -> TDouble, "Cov2" -> TDouble)))
+
+    val (linregKT, sampleKT) = vdsBurdenNoOverlap.linregBurden("gene", "va.gene",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1"),
+      singleKey = true)
+
+    val (linregKT2, sampleKT2) = vdsBurdenNoOverlap.linregBurden("gene", "va.genes2",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1"))
+
+    val (linregKT3, sampleKT3) = vdsBurdenNoOverlap.linregBurden("gene", "va.genes3",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1"))
+
+    val (linregKT4, sampleKT4) = vdsBurdenNoOverlap.linregBurden("gene", "va.genes4",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1"))
+
+    val (linregKT5, sampleKT5) = vdsBurdenNoOverlap.linregBurden("gene", "va.genes5",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1"))
+
+    val (linregKT6, sampleKT6) = vdsBurdenNoOverlap.linregBurden("gene", "va.genes6",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1"))
+
+    val (linregKT7, sampleKT7) = vdsBurdenNoOverlap.linregBurden("gene", "va.genes7",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1"),
+      singleKey = true)
+
+    val (linregKT8, sampleKT8) = vdsBurdenNoOverlap.linregBurden("gene", "va.genes8",
+      "gs.map(g => va.weight * g.gt).sum()", "sa.pheno.Pheno", covariates = Array("sa.cov.Cov1"),
+      singleKey = true)
```

Comment:
Can we add a test with a covariate that is an expression that isn't also an annotation path?